server/api.js

```diff
@@ -135,116 +147,118 @@ router.post("/modules/textBanner/:pageTitle", async (req, res) => {
-		});
-	} catch (error) {
-		console.error(error);
-		
 		res.status(500).json({ error });
 	}
 });
 
-router.post("/modules/heroBanner/:pageTitle", async (req, res) => {
-	const pageTitle = req.params.pageTitle;
-	const { heroImage, heroText } = req.body;
 
-	if (!heroImage ||!heroText ) {
-		res.status(500).json("Please fill all the fields");
+router.post("/modules/imageAndTexts/:pageTitle",async (req, res) => {
+		const pageTitle = req.params.pageTitle;
+		const {
+			text_header,
+			text_body,
+			image,
+			button,
+			hasbutton,
+			imagetext_direction,
+		} = req.body;
+
+		if (
+			!text_header ||
+			!text_body ||
+			!image ||
+			!button
+		) {
+			res.status(500).json("Please fill all the fields");
+		}
+
+		try {
+			const findPageId = await db.query(
+				`SELECT page_id FROM pages WHERE page_title = $1`,
+				[pageTitle]
+			);
+			const page_id = findPageId.rows[0].page_id;
+
+			const imageAndTexts = await db.query(
+				`insert into imageandtexts (text_header, text_body, image, button, hasbutton, imagetext_direction) 
+       VALUES ($1, $2, $3, $4, $5, $6) returning record_id`,
+				[
+					text_header,
+					text_body,
+					image,
+					button,
+					hasbutton,
+					imagetext_direction,
+				]
+			);
+
+
+			const record_id = imageAndTexts.rows[0].record_id;
+
+			const insertingModulesTable = await db.query(
+				`INSERT INTO modules (page_id, module_type, record_id) VALUES ($1, $2, $3)`,
+				[page_id, "imageAndTexts", record_id]
+			);
+
+			return res.status(200).json({
+				message: `Data is stored successfully into imageAndTexts table`,
+			});
+		} catch (error) {
+			console.error(error);
+
+			res.status(500).json({ error });
+		}
 	}
-
-	try {
-		
-		// First inserting the textBanner table
-		const imageAndTexts = await db.query(
-			`INSERT INTO herobanner (hero_image,hero_text) 
-      VALUES ($1, $2)`,
-			[heroImage, heroText]
-		);
-
-		//Second finding the pagge id
-		const findPageId = await db.query(
-			`select page_id from pages where page_title = $1`,
-			[pageTitle]
-		);
-		const page_id = findPageId.rows[0].page_id;
-		console.log(page_id);
-
-		// third finding the last record in the textBnner table
-		const lastRecord = await db.query(
-			`select record_id from herobanner order by record_id desc limit 1`
-		);
-		const record_id = lastRecord.rows[0].record_id;
-
-		// fourth inserting the above datas into the modules table
-		const insertingModulesTable = await db.query(
-			`insert into modules(page_id,module_type,record_id) values($1,$2,$3)`,
-			[page_id, "heroBanner", record_id]
-		);
-		
-		return res.status(200).json({
-			message: `Data is stored successfuly into imageAndTexts table`,
-		});
-	} catch (error) {
-		console.error(error);
-		
-		res.status(500).json({ error });
+);
+
+router.post("/modules/heroBanner/:pageTitle",async (req, res) => {
+		const pageTitle = req.params.pageTitle;
+		const { heroImage, heroText } = req.body;
+
+
+		console.log("====>>>", heroText);
```

Comment:
Well done @khmdagal, 
 just if you can remove console logs

Reply:
Sure